### PR TITLE
JAMES-1888 Cassandra read should not fail on reading mail with attachment

### PR DESCRIPTION
**Pull request by chibenwa**
When reading mails with IMAP, trying to introduce the DEMO, I have : 

```
05:07:18.282 [WARN ] o.a.j.m.p.IMAPServerModule - ID=-560007929 Error while processing imap request: java.lang.IllegalArgumentException - Multiple entries with same key: AttachmentId{id=3faab827131a39bfcba6b4a2f7dbbbb3e310a37b}=MessageAttachmentRepresentation{attachmentId=AttachmentId{id=3faab827131a39bfcba6b4a2f7dbbbb3e310a37b}, name=Optional[meeting.ics], cid=Optional.empty, isInline=false} and AttachmentId{id=3faab827131a39bfcba6b4a2f7dbbbb3e310a37b}=MessageAttachmentRepresentation{attachmentId=AttachmentId{id=3faab827131a39bfcba6b4a2f7dbbbb3e310a37b}, name=Optional[Portion de message joint], cid=Optional.empty, isInline=false}
```

This is due to : 

```
java.lang.IllegalArgumentException: Multiple entries with same key: AttachmentId{id=3faab827131a39bfcba6b4a2f7dbbbb3e310a37b}=MessageAttachmentRepresentation{attachmentId=AttachmentId{id=3faab827131a39bfcba6b4a2f7dbbbb3e310a37b}, name=Optional[meeting.ics], cid=Optional.empty, isInline=false} and AttachmentId{id=3faab827131a39bfcba6b4a2f7dbbbb3e310a37b}=MessageAttachmentRepresentation{attachmentId=AttachmentId{id=3faab827131a39bfcba6b4a2f7dbbbb3e310a37b}, name=Optional[Portion de message joint], cid=Optional.empty, isInline=false}
    at com.google.common.collect.ImmutableMap.checkNoConflict(ImmutableMap.java:150)
    at com.google.common.collect.RegularImmutableMap.checkNoConflictInBucket(RegularImmutableMap.java:104)
    at com.google.common.collect.RegularImmutableMap.<init>(RegularImmutableMap.java:70)
    at com.google.common.collect.ImmutableMap$Builder.build(ImmutableMap.java:254)
    at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:503)
    at org.apache.james.mailbox.cassandra.mail.AttachmentLoader.getAttachments(AttachmentLoader.java:47)
```

This commit solves the issue.

### CODE REVIEW

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java [hunk 1]
```diff
@@ -321,7 +321,7 @@ private SharedByteArrayInputStream buildContent(Row row, FetchType fetchType) {
         return headerContent;
     }
 
-    static class MessageAttachmentRepresentation {
+    @VisibleForTesting public static class MessageAttachmentRepresentation {
```
Comment (aduprat) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java]: If it's public you don't need @VisibleForTesting.

File: mailbox/api/src/main/java/org/apache/james/mailbox/model/MessageAttachment.java [hunk 2]
```diff
@@ -88,7 +88,7 @@ public MessageAttachment build() {
     private final Optional<Cid> cid;
     private final boolean isInline;
 
-    @VisibleForTesting MessageAttachment(Attachment attachment, Optional<String> name, Optional<Cid> cid, boolean isInline) {
+    @VisibleForTesting public MessageAttachment(Attachment attachment, Optional<String> name, Optional<Cid> cid, boolean isInline) {
```
Comment (aduprat) [inline mailbox/api/src/main/java/org/apache/james/mailbox/model/MessageAttachment.java]: If it's public you don't need @VisibleForTesting.

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java (hunk 1 above)
Comment (chibenwa) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageDAO.java]: Ok

File: mailbox/api/src/main/java/org/apache/james/mailbox/model/MessageAttachment.java (hunk 2 above)
Comment (chibenwa) [inline mailbox/api/src/main/java/org/apache/james/mailbox/model/MessageAttachment.java]: Ok

**Comment by chibenwa** [comment]
test this please

**Comment by rouazana** [comment]
test this please

**Comment by rouazana** [comment]
test this please

**Pull request merged**